subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/CompositeDependencyArtifactsVisitor.java

```diff
@@ -19,15 +19,14 @@
 import com.google.common.collect.ImmutableList;
 import org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.ArtifactSet;
 import org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.DependencyArtifactsVisitor;
-import org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.ValidatingArtifactsVisitor;
 import org.gradle.internal.component.local.model.LocalFileDependencyMetadata;
 
 import java.util.List;
 
-public class CompositeDependencyArtifactsVisitor implements ValidatingArtifactsVisitor {
+public class CompositeDependencyArtifactsVisitor implements DependencyArtifactsVisitor {
     private final List<DependencyArtifactsVisitor> visitors;
 
-    public CompositeDependencyArtifactsVisitor(List<DependencyArtifactsVisitor> visitors) {
+    public CompositeDependencyArtifactsVisitor(DependencyArtifactsVisitor... visitors) {
```

Comment:
💭 Minor thing, but I usually prefer to avoid varargs and take a list, just because arrays are mutable and a little janky. It doesn't matter that much here, though.

Reply:
Will revert to a list